### ISSUE TITLE
Pings with missing installation parameter sent to ET

### ISSUE THREAD

**Issue by btepe** [camunda/camunda-modeler#2076]
__Describe the Bug__

[ET](https://github.com/camunda/et) has rejected over 1000 pings from the modeler over the last 5 days (the first one was on 18.01.21 at 11:39am), because of a missing "installation" parameter (see [Google cloud error logs](https://console.cloud.google.com/logs/query;query=error_group%2528%22CM_Gor_G48XDcw%22%2529%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22sharedservicesproduction%22%0Aresource.labels.namespace_name%3D%22et-production%22%0Aresource.labels.container_name%3D%22rest%22%0AjsonPayload.message%3D%22middlewareError%20for%20request%20from%20Camunda%20BPM%20Runtime%20null%20community:%20ValidationError:%20%5C%22product.version%5C%22%20must%20be%20a%20string%22;timeRange=2021-01-06T12:58:03.137Z%2F2021-01-22T12:58:03.137653Z?folder=true&organizationId=669107107215&project=shared-services-camunda-cloud)).

This rejected data is not showing in Kibana and might distort your analysis there, so you should probably look into why this is happening and how it can be fixed.

__Steps to Reproduce__

none

__Expected Behavior__

The pings sent to ET are expected to include the "installation" parameter.

__Environment__

- OS: probably irrelevant
- Camunda Modeler Version: 4.5.0 Community

/cc @casulond 

Related to https://github.com/camunda/camunda-modeler/issues/1965

**Comment by pinussilvestrus** [comment]
Thanks for reporting!

We had the same problem in the past (https://github.com/camunda/camunda-modeler/issues/1965), but seems like it was hard to reproduce and not easy to fix. We refactored the update server a lot in the last few days (cc @nikku), so maybe (hopefully) we fixed this problem as well on the road 🤞 

We should keep on having our eyes on that.

**Comment by casulond** [comment]
CC @quentinleroy

Just for awareness, no action required

**Comment by pinussilvestrus** [comment]
> because of a missing "product.version" parameter (see [Google cloud error logs](https://console.cloud.google.com/logs/query;query=error_group%2528%22CM_Gor_G48XDcw%22%2529%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22sharedservicesproduction%22%0Aresource.labels.namespace_name%3D%22et-production%22%0Aresource.labels.container_name%3D%22rest%22%0AjsonPayload.message%3D%22middlewareError%20for%20request%20from%20Camunda%20BPM%20Runtime%20null%20community:%20ValidationError:%20%5C%22product.version%5C%22%20must%20be%20a%20string%22;timeRange=2021-01-06T12:58:03.137Z%2F2021-01-22T12:58:03.137653Z?folder=true&organizationId=669107107215&project=shared-services-camunda-cloud)).

I was trying to monitor this. But seems like I have not the necessary permissions

![image](https://user-images.githubusercontent.com/9433996/105835776-8ee0db00-5fcc-11eb-985b-69f0e6e693bb.png)

I am using my Camunda account on Google. Is there anything else I need to have?

**Comment by casulond** [comment]
> > because of a missing "product.version" parameter (see [Google cloud error logs](https://console.cloud.google.com/logs/query;query=error_group%2528%22CM_Gor_G48XDcw%22%2529%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22sharedservicesproduction%22%0Aresource.labels.namespace_name%3D%22et-production%22%0Aresource.labels.container_name%3D%22rest%22%0AjsonPayload.message%3D%22middlewareError%20for%20request%20from%20Camunda%20BPM%20Runtime%20null%20community:%20ValidationError:%20%5C%22product.version%5C%22%20must%20be%20a%20string%22;timeRange=2021-01-06T12:58:03.137Z%2F2021-01-22T12:58:03.137653Z?folder=true&organizationId=669107107215&project=shared-services-camunda-cloud)).
> 
> I was trying to monitor this. But seems like I have not the necessary permissions
> 
> ![image](https://user-images.githubusercontent.com/9433996/105835776-8ee0db00-5fcc-11eb-985b-69f0e6e693bb.png)
> 
> I am using my Camunda account on Google. Is there anything else I need to have?

We're sorry for the information about the missing `product.version` in some parts of the issue description. It is always the `installation` that is missing. That was a typo - I corrected the initial comment.

I'd like to share a log message with you. We should have done that initially instead of sharing a link because we don't want to grant access to the logs in general. I'm sharing a snippet of an error log entry with you - we log only limited fields from the request:

```json
{
  "message": "ValidationError: \"installation\" is required for request from Camunda Modeler 4.5.0 Community (middlewareError)",
  "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
  "meta": {
    "stack": "ValidationError: \"installation\" is required",
    "message": "uncaughtException: \"installation\" is required\nValidationError: \"installation\" is required",
    "exception": true,
    "level": "error",
    "error": {
      "statusCode": 422,
      "expose": true,
      "details": [
        {
          "type": "any.required",
          "message": "\"installation\" is required",
          "path": [
            "installation"
          ],
          "context": {
            "key": "installation",
            "label": "installation"
          }
        }
      ],
      "status": 422
    },
    "date": "Tue Jan 26 2021 10:40:40 GMT+0000 (Coordinated Universal Time)",
    "trace": [],
    "req": {
      "body": {
        "product": {
          "name": "Camunda Modeler",
          "edition": "Community",
          "version": "4.5.0",
          "internals": {
            "event": "ping"
          }
        }
      },
      "method": "POST",
      "httpVersion": "1.1",
      "headers": {
        "accept": "application/json",
        "content-length": "107",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) CamundaModeler/4.5.0 Chrome/78.0.3904.130 Electron/7.1.14 Safari/537.36",
        "content-type": "application/json",
        "host": "api.telemetry.camunda.cloud"
      }
    }
  }
}
```

Let me know if you need more samples or other information. We can also talk on Zoom or so and have a look at the logs together with screen-sharing.

**Comment by nikku** [comment]
Proposal: Before sending the PING we validate that the installation is set on our side. If not, that is an error we track in Sentry. This way we can advance this topic on our side (and fix the broken behavior, eventually).

**Comment by casulond** [comment]
Above, GitHub mentioned milestone M45 in relation to this issue:

![image](https://user-images.githubusercontent.com/60668314/108819150-c0909600-75ba-11eb-8515-c0188516c165.png)

Looking at the [milestone M45](https://github.com/camunda/camunda-modeler/milestone/57), it is closed while this issue is still open. What is your targeted schedule for fixing this issue?

**Comment by pinussilvestrus** [comment]
tbh: I would just go with [the solution @nikku proposed](https://github.com/camunda/camunda-modeler/issues/2076#issuecomment-769444345)

* keep the broken data away from E.T.
* track it our side, where it has less impact

I'll follow up on this.

**Comment by nikku** [comment]
@casulond The issue is not part of the milestone anymore. Hence it is still open.

**Comment by pinussilvestrus** [comment]
We were able to find the problem: https://github.com/camunda/camunda-modeler/pull/2116

tldr: the installation id is only set on early enabling (on application start). If you enable the usage statistics on runtime, e.g. via the privacy preferences modal, the editor id would be empty. So it's likely that there is a missing editor id in the E.T. payload _for every user who enables E.T. as a first step_.

The PR should fix this problem that we don't send a missing installation id to E.T. + making sure it's available anytime soon.

Thanks a lot for your patience @btepe @casulond 👍

**Comment by nikku** [comment]
Great work guys :clap:. And thanks for being persistent @btepe @casulond, too :tada:.

**Comment by btepe** [comment]
Great that you found the source of the issue 👏 🎉

**Comment by nikku** [comment]
Closed via https://github.com/camunda/camunda-modeler/pull/2116.
